app/javascript/components/patient/vaccines/VaccineModal.js

```diff
@@ -0,0 +1,244 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Button, Modal, Row, Col, Form } from 'react-bootstrap';
+import Select from 'react-select';
+import moment from 'moment';
+import ReactTooltip from 'react-tooltip';
+
+import DateInput from '../../util/DateInput';
+
+class VaccineModal extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      product_name: this.props.currentVaccineData?.product_name,
+      // If there is not already a group name, default to the first option
+      group_name: this.props.currentVaccineData?.group_name
+        ? this.props.currentVaccineData.group_name
+        : this.props.group_name_options
+        ? this.props.group_name_options[0]
+        : null,
+      administration_date: this.props.currentVaccineData?.administration_date,
+      dose_number: this.props.currentVaccineData?.dose_number,
+      notes: this.props.currentVaccineData?.notes,
+      // Sorting so that the blank option shows up at the top
+      sorted_dose_number_options: this.props.dose_number_options ? this.props.dose_number_options.sort() : [],
+      isValid: false,
+    };
+  }
+
+  componentDidMount() {
+    this.checkIfValid();
+  }
+
+  /**
+   * Data is valid and can be saved if there is both a group name and product name selected.
+   */
+  checkIfValid = () => {
+    const isValid = this.state.group_name && this.state.product_name;
```

Comment:
should we put an asterisk next to the fields that are required to make it more immediately obvious to the users that they are required like we do in other places in the app?

Reply:
I agree with this, I was initially confused when testing the disabling of the submit button.  I also noticed we didn't do this with the Close Contact modal either, but there's more either or logic with that one

Reply:
Yep, absolutely - meant to do this and it slipped my mind. Good catch - will add soon.

Reply:
Added here: https://github.com/SaraAlert/SaraAlert/pull/721/commits/7b5ec1bd59c6c8c33fdb9331c5cf35933767c972#diff-182e13a82e02db14eb5222d1eb289b14639dc17acd927608ff7e8b032cfb9af8R81